tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterBase.java

```diff
@@ -1,289 +1,151 @@
-import org.eclipse.hono.client.MessageSender;
 import org.eclipse.hono.client.impl.HonoClientImpl;
 import org.eclipse.hono.config.ClientConfigProperties;
 import org.eclipse.hono.tests.IntegrationTestSupport;
-import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.tests.client.ClientTestBase;
 import org.eclipse.hono.util.TenantObject;
 import org.junit.After;
 import org.junit.AfterClass;
-import org.junit.Before;
 import org.junit.BeforeClass;
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.TestName;
-import org.junit.rules.Timeout;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
+import io.vertx.core.CompositeFuture;
 import io.vertx.core.Future;
-import io.vertx.core.Vertx;
 import io.vertx.ext.unit.Async;
 import io.vertx.ext.unit.TestContext;
-import io.vertx.proton.ProtonClientOptions;
 import io.vertx.proton.ProtonHelper;
 
 /**
  * Base class for the AMQP Adapter integration tests.
  */
-public abstract class AmqpAdapterBase {
+public abstract class AmqpAdapterBase extends ClientTestBase {
 
-    private final Logger LOG = LoggerFactory.getLogger(getClass());
+    private static final String AMQP_ADPATER_TEST_DEVICE = "amqp-adapter-device";
 
-    private static final int MESSAGES_TO_SEND = 200;
+    private static final String AMQP_ADAPTER_TEST_TENANT = "amqp-adapter-tenant";
 
-    private static final int TEST_TIMEOUT = 10000;
-
-    @Rule
-    public final Timeout timeout = new Timeout(TEST_TIMEOUT, TimeUnit.MILLISECONDS);
-
-    @Rule
-    public final TestName testName = new TestName();
+    private static final String DEVICE_PWD = "Ddevice-password";
 
     /**
-     * The Vert.x instance to run on.
-     */
-    private static Vertx vertx;
-
-    private static final Set<HonoClient> clientsToShutdown = new HashSet<>();
-
-    /**
-     * A helper for accessing the AMQP 1.0 messaging network and for managing devices tenants, devices and credentials.
-     */
-    protected static IntegrationTestSupport helper;
-
-    protected abstract Future<MessageConsumer> createConsumer(String tenantId, Consumer<Message> messageConsumer);
-
-    protected abstract Future<MessageSender> getOrCreateSender(String tenantId, HonoClient client);
-
-    /**
-     * Sets up and initializes the integration test setup.
+     * A setup that initializes a HTTP endpoint for accessing the device registry.
+     * It also creates a test tenant and registers a device (with credentials)
      * 
-     * @param context The Vert.x test context.
+     * @param ctx The vertx test context.
      */
     @BeforeClass
-    public static void init(final TestContext context) {
-        vertx = Vertx.vertx();
+    public static void init(final TestContext ctx) {
         helper = new IntegrationTestSupport(vertx);
-        helper.init(context);
+        helper.initRegistryClient(ctx);
+
+        final Async setup = ctx.async();
+        final TenantObject tenant = TenantObject.from(AMQP_ADAPTER_TEST_TENANT, true);
```

Comment:
I would rather have the individual tests register random tenants and devices ...

Reply:
thanks. will work on this now